### PR TITLE
ethkey and ethstore use hash structures from bigint

### PR DESCRIPTION
**Pull request by debris**
in next pr, I will remove duplicated signing logic (it's in ethkey and util/crypto)

### CODE REVIEW

**Comment by rphmeier** [comment]
needs a merge

File: util/src/lib.rs [hunk 1]
```diff
@@ -156,59 +156,5 @@ pub use log::*;
 pub use kvdb::*;
 pub use timer::*;
 
-#[cfg(test)]
```
Comment (NikVolf) [inline util/src/lib.rs]: how is it related to the topic of the pr?

Comment (debris) [inline util/src/lib.rs]: ahh... you are right. this test was just moved to different file `util/bigint/src/uint.rs', cause it fits there better

File: json/src/hash.rs [hunk 2]
```diff
@@ -19,7 +19,7 @@
 use std::str::FromStr;
 use serde::{Deserialize, Deserializer, Error};
 use serde::de::Visitor;
-use util::hash::{H64 as Hash64, Address as Hash160, H256 as Hash256, H2048 as Hash2048};
+use util::hash::{H64 as Hash64, H160 as Hash160, H256 as Hash256, H2048 as Hash2048};
```
Comment (gavofyork) [inline json/src/hash.rs]: prefer to stick to `Address` over `H160` as long as it's being used for an address.

File: rpc/src/v1/types/hash.rs [hunk 3]
```diff
@@ -20,7 +20,7 @@ use std::cmp::Ordering;
 use std::hash::{Hash, Hasher};
 use serde;
 use rustc_serialize::hex::{ToHex, FromHex};
-use util::{H64 as Eth64, H256 as EthH256, H520 as EthH520, H2048 as Eth2048, Address};
+use util::{H64 as Eth64, H256 as EthH256, H520 as EthH520, H2048 as Eth2048, H160 as Eth160};
```
Comment (gavofyork) [inline rpc/src/v1/types/hash.rs]: Same

**Comment by gavofyork** [comment]
not massively keen on the error mapping introduced, but at least there's a lot of other boilerplate removed.

as for the types, unless we're actually using a 160-bit datum for something generic, then it should be named `Address`. same goes for `H520` and `Signature` and `H256` and `Secret`.

**Comment by debris** [comment]
> unless we're actually using a 160-bit datum for something generic, then it should be named `Address`. same goes for `H520` and `Signature` and `H256` and `Secret`.

I agree. I just renamed usages of `Address` to `H160` in bigint library and every place where it's not using address is not specific (eg. serialization).

**Pull request merged**